tests/tools/src/cpu/CMakeLists.txt

```diff
@@ -32,22 +32,65 @@ detray_add_executable(generate_telescope_detector
 # Build the visualization executable.
 detray_add_executable(detector_display
                       "detector_display.cpp"
-                      LINK_LIBRARIES Boost::program_options detray::tools
+                      LINK_LIBRARIES Boost::program_options  detray::core_array detray::tools
                       detray::svgtools
 )
 
 # Build the detector validation executable.
 detray_add_executable(detector_validation
                       "detector_validation.cpp"
                       LINK_LIBRARIES GTest::gtest GTest::gtest_main
-                      Boost::program_options detray::tools detray::test_utils
+                      Boost::program_options detray::core_array detray::tools detray::test_utils
                       detray::svgtools
 )
 
 # Build the material validation executable.
 detray_add_executable(material_validation
                       "material_validation.cpp"
                       LINK_LIBRARIES GTest::gtest GTest::gtest_main
-                      Boost::program_options detray::tools detray::test_utils
+                      Boost::program_options detray::core_array detray::tools detray::test_utils
                       detray::svgtools
 )
+
+if(DETRAY_BUILD_BENCHMARKS)
+    # Look for openMP, which is used for the CPU propagation benchmark
+    find_package(OpenMP)
+
+    # Build the propagation benchmark executable.
+    macro(detray_add_propagation_benchmark algebra)
+        detray_add_executable(propagation_benchmark_cpu_${algebra}
+                            "propagation_benchmark.cpp"
+                            LINK_LIBRARIES Boost::program_options benchmark::benchmark benchmark::benchmark_main vecmem::core detray::benchmark_cpu detray::core_${algebra} detray::tools detray::detectors
+        )
+
+        target_compile_options(
+            detray_propagation_benchmark_cpu_${algebra}
+            PRIVATE "-march=native" "-ftree-vectorize"
+        )
```

Comment:
This should probably not be hardcoded by the detray build system but rather taken from the `CXXFLAGS` environment variable.